hnn_core/cell.py

```diff
@@ -361,14 +419,48 @@ def _set_biophysics(self, sections):
             for mech_name, p_mech in section.mechs.items():
                 sec.insert(mech_name)
                 for attr, val in p_mech.items():
-                    if hasattr(val, '__call__'):
-                        sec.push()
-                        for seg in sec:
-                            setattr(seg, attr, val(h.distance(seg.x)))
-                        h.pop_section()
+                    if isinstance(val, list):
+                        seg_xs, seg_vals = val[0], val[1]
+                        for seg, seg_x, seg_val in zip(sec, seg_xs, seg_vals):
+                            # Checking equaliy till 5 decimal places
+                            np.testing.assert_almost_equal(seg.x, seg_x, 5)
+                            setattr(seg, attr, seg_val)
```

Comment:
let's be brave and delete `_set_biophysics`

Reply:
Don't we still need this function for inserting nrn mechanisms and such?

Reply:
I think `_set_section_mechs` is doing that now. This method only exists for checking equality with the old stuff

Reply:
I think it is still required as it is inserting mechanisms in the NEURON section objects.
`_set_section_mechs()` is just expanding the partial functions for now. If we include the `_set_biophysics()` logic here then NEURON objects might still be created before simulating the network (in case of network write)